app/src/main/java/dev/arkbuilders/arkmemo/ui/fragments/NotesFragment.kt

```diff
@@ -65,51 +61,79 @@ class NotesFragment: Fragment(R.layout.fragment_notes) {
+                    binding.rvPinnedNotes.adapter?.notifyItemChanged(deletePosition)
+            }).show(childFragmentManager, CommonActionDialog.TAG)
+        }
+
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        binding = FragmentHomeBinding.inflate(layoutInflater)
         notesViewModel.apply {  init { readAllNotes() } }
     }
 
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentHomeBinding.inflate(layoutInflater)
+        return binding.root
+    }
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        val fabNewVoiceNote = binding.fabNewVoiceNote
-        recyclerView = binding.include.recyclerView
         activity.title = getString(R.string.app_name)
         activity.supportActionBar?.setDisplayHomeAsUpEnabled(false)
-        newTextNoteButton = binding.newTextNote
-        pasteNoteButton = binding.pasteNote
-        newGraphicNoteButton = binding.newGraphicNote
-        newNoteButton = binding.newNote
-        newNoteButton.shrink()
-        showFabs = false
-        newNoteButton.setOnClickListener {
-            showFabs = if (!showFabs) {
-                newNoteButton.extend()
-                newTextNoteButton.show()
-                newGraphicNoteButton.show()
-                fabNewVoiceNote.show()
-                true
-            } else {
-                newNoteButton.shrink()
-                newTextNoteButton.hide()
-                newGraphicNoteButton.hide()
-                fabNewVoiceNote.hide()
-                false
-            }
-        }
-        newTextNoteButton.setOnClickListener(newTextNoteClickListener)
-        newGraphicNoteButton.setOnClickListener(newGraphicNoteClickListener)
-        pasteNoteButton.setOnClickListener(pasteNoteClickListener)
-        fabNewVoiceNote.setOnClickListener {
-            activity.fragment = ArkRecorderFragment.newInstance()
-            activity.replaceFragment(activity.fragment, ArkRecorderFragment.TAG)
+        binding.ivSettings.setOnClickListener {
+            activity.fragment = SettingFragmentsV2()
+            activity.replaceFragment(activity.fragment, SettingFragmentsV2::class.java.name)
         }
+        showingFloatingButtons = false
+        initBottomControlViews()
         lifecycleScope.launchWhenStarted {
             notesViewModel.getNotes {
+                notes = it
                 val adapter = NotesListAdapter(
                     it,
                     onPlayPauseClick = { path ->
                         arkMediaPlayerViewModel.onPlayOrPauseClick(path)
+                    },
+                    onThumbPrepare = { graphicNote, noteCanvas ->
+//                        val tempNoteViewModel: GraphicNotesViewModel by viewModels()
```

Comment:
I have observed, the graphic notes crash comes from here. because we are not calling ```noteCanvas.setViewModel(viewModel: GraphicNotesViewModel)``` here. I guess this function is empty intentionally, is it?,

Reply:
@shuberttheI From the stacktrace, yes this seems to be the cause.
However, it's surprising that it's not happening in my device even though I also have several graphic notes here.
Could you provide more info about your device? (Model name/ OS version) 
Or any other special conditions if you think they could be related.

Thanks~

Reply:
Model HUAWEI SNE-LX1
Android 10

Reply:
@shuberttheI Yes, it's happening in Android 10.
Can you verify the crash again with latest changes please?